src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/TargetFrameworkMonikersValueProvider.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using static Microsoft.VisualStudio.ProjectSystem.Build.TargetFrameworkProjectConfigurationDimensionProvider;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties.InterceptedProjectProperties
+{
+    [ExportInterceptingPropertyValueProvider("TargetFrameworkMonikers", ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+    internal sealed class TargetFrameworkMonikersValueProvider : InterceptingPropertyValueProviderBase
+    {
+        private readonly ProjectProperties _properties;
+
+        [ImportingConstructor]
+        public TargetFrameworkMonikersValueProvider(ProjectProperties properties)
+        {
+            _properties = properties;
+        }
+
+        public override async Task<string> OnGetEvaluatedPropertyValueAsync(string evaluatedPropertyValue, IProjectProperties defaultProperties)
+        {
+            var configuration = await _properties.GetConfigurationGeneralPropertiesAsync().ConfigureAwait(true);
+            var currentTargetFramework = (string)await configuration.TargetFramework.GetValueAsync().ConfigureAwait(true);
+            var currentTargetFrameworks = (string)await configuration.TargetFrameworks.GetValueAsync().ConfigureAwait(true);
+            if (!string.IsNullOrEmpty(currentTargetFrameworks))
+            {
+                // sorting the target frameworks so projects that specify the same set of frameworks return the same string
+                return string.Join(";", ParseTargetFrameworks(currentTargetFrameworks).Sort());
```

Comment:
Where is ParseTargetFrameworks defined? 

We're not supposed to be reading the TargetFrameork or TargetFrameworks properties - these are user defined properties with user defined values, I'm actively reducing our dependency on these properties within the project, they aren't supposed to be interpreted - that's the build's job.

Instead, we should be reading the TargetFrameworkMoniker property from all active configs which already has the parsed moniker or explicitly specified moniker.